homeassistant/components/xbox/config_flow.py

```diff
@@ -31,9 +38,23 @@ async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Handle a flow start."""
-        await self.async_set_unique_id(DOMAIN)
 
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
 
         return await super().async_step_user(user_input)
+
+    async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
+        """Create an entry for the flow."""
+
+        async with SignedSession() as session:
+            auth = AuthenticationManager(session, "", "", "")
+            auth.oauth = OAuth2TokenResponse(**data["token"])
+            await auth.refresh_tokens()
+
+            client = XboxLiveClient(auth)
+
+            me = await client.people.get_friends_own_batch([client.xuid])
+            await self.async_set_unique_id(client.xuid)
```

Comment:
This should be done outside of the managed context.

../Frenck

```suggestion
        await self.async_set_unique_id(client.xuid)
```